src/lib/decode.ts

```diff
@@ -0,0 +1,321 @@
+        // AVS writes out 0 in that case, and sets the actual id at the end of the save block.
+        const effectIdOld = get.UInt32(blob, offset);
+        let effect = [];
+        let code;
+        let hidden: string[];
+        if (effectIdOld !== 0) {
+            if (effectIdOld === 0x7fff) {
+                let strAndSize: [string, number] | [string, number, string[]] = ['', 0];
+                if (blob[offset + config.sizeInt] === 1) {
+                    // new-version marker
+                    strAndSize = get.SizeString(blob, offset + config.sizeInt + 1);
+                } else {
+                    strAndSize = get.SizeString(blob, offset + config.sizeInt, 256);
+                }
+                offset += strAndSize[1];
+                code = strAndSize[0];
+                if (strAndSize.length > 2) {
+                    hidden = (<[string, number, string[]]>strAndSize)[2];
+                }
+            } else {
+                if (effectIdOld > 15) {
+                    if (verbosity >= 0) {
+                        Log.error(`Movement: Unknown effect id ${effectIdOld}. This is a known bug.`);
+                        console.log('If you know an AVS version that will display this Movement as anything else but "None", then please send it in!');
+                    }
+                    effect = Table.movementEffect[0];
+                } else {
+                    effect = Table.movementEffect[effectIdOld];
+                }
+            }
+        } else {
+            let effectIdNew = 0;
+            if (offset + config.sizeInt * 6 < end) {
+                effectIdNew = get.UInt32(blob, offset + config.sizeInt * 6); // 1*config.sizeInt, because of oldId=0, and 5*config.sizeint because of the other settings.
+            }
+            effect = Table.movementEffect[effectIdNew];
+        }
+        if (effect && effect.length > 0) {
+            comp['builtinEffect'] = effect[0];
+        }
+        comp['output'] = get.UInt32(blob, offset + config.sizeInt) ? '50/50' : 'Replace';
+        comp['sourceMapped'] = get.Bool(blob, offset + config.sizeInt * 2, config.sizeInt)[0];
+        comp['coordinates'] = Table.coordinates[get.UInt32(blob, offset + config.sizeInt * 3)];
+        comp['bilinear'] = get.Bool(blob, offset + config.sizeInt * 4, config.sizeInt)[0];
+        comp['wrap'] = get.Bool(blob, offset + config.sizeInt * 5, config.sizeInt)[0];
+        if (effect && effect.length && effectIdOld !== 1 && effectIdOld !== 7) {
+            // 'slight fuzzify' and 'blocky partial out' have no script representation.
+            code = effect[1];
+            comp['coordinates'] = effect[2]; // overwrite
+        }
+        comp['code'] = code;
+        if (hidden) {
+            comp['_hidden'] = hidden;
+        }
+        return comp;
+    },
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    avi(blob: Uint8Array, offset: number): unknown {
```

Comment:
I think the linter-disable line is not needed here? Looks like all parameters are used.

Reply:
fixed in 023acc0